File: course_materials/linguistics/python/load.py
import pandas as pd


def load_q_and_a_key(path = "../data"):
    """
    Load the question and answer key from the given path.
    """
    # Load the question and answer key
    q_and_a_key = pd.read_csv(f"{path}/q_and_a_key.csv")
    return q_and_a_key


def load_ling_data(path = "../data"):
    """
    Load the linguistic data from the given path.
    """
    # Load the linguistic data
    ling_data = pd.read_csv(
        f"{path}/ling_data.csv",
        # make explicit that the ZIP column should be read as a character
        # # (default would've been to load ZIP in as numeric; this may drop the leading 0s)
        dtype={"ZIP": str}
    )
    return ling_data

File: course_materials/linguistics/R/load.R
#' Load in question and answer key
#'
#' @param path the path to the data
#'
#' @returns A data frame containing the question and answer key
load_q_and_a_key <- function(path = here::here("data")) {
  readr::read_csv(
    file.path(path, "q_and_a_key.csv"),
    show_col_types = FALSE
  ) |>
    tibble::as_tibble()
}


#' Load in lingusitics data
#'
#' @param path the path to the data
#'
#' @returns A data frame containing the linguistics survey data
load_ling_data <- function(path = here::here("data")) {
  readr::read_csv(
    file.path(path, "ling_data.csv"),
    col_types = readr::cols(
      ZIP = readr::col_character()
    ),
    show_col_types = FALSE
  ) |>
    tibble::as_tibble()
}
